### ISSUE TITLE
Media: Close attachment details modal with esc key

### ISSUE THREAD

**Issue by viktorix** [ClassicPress/ClassicPress#835]
This is a backport: https://core.trac.wordpress.org/changeset/51945
Relevant ticket: https://core.trac.wordpress.org/ticket/53924

> The event that fired closing the attachment details modal also removed the keydown event listener, so subsequent modals could not be closed with the escape key.

## Expected behavior
Subsequent media modals should be closed with the escape key.

## Current behavior
The event that fired closing the attachment details modal also removed the keydown event listener, so subsequent modals could not be closed with the escape key.

## Possible solution
Backport https://core.trac.wordpress.org/changeset/51945

## Context
Accessibility backport.

**Comment by viktorix** [comment]
With re-fork work getting underway, we should consider closing backporting issues such as this.

**Comment by viktorix** [comment]
Will be included in v2.0.
